drv/stm32h7-update-server/src/main.rs

```diff
@@ -163,13 +163,23 @@ impl<'a> ServerImpl<'a> {
     ) -> Result<(), RequestError<UpdateError>> {
         ringbuf_entry!(Trace::WriteStart);
 
-        if word_number > BANK_WORD_LIMIT {
+        // These variables are _philosophically_ constants, but since they're
+        // generated by taking the address of a linker-generated symbol, we
+        // can't define them as `const` values.
+        //
+        // SAFETY: these are symbols populated by the linker.
+        let bank_addr = unsafe { __REGION_BANK2_BASE.as_ptr() } as u32;
+        let bank_end = unsafe { __REGION_BANK2_END.as_ptr() } as u32;
```

Comment:
Tiny nit - would `as usize` here prevent the `as u32` casts below it?

Reply:
Good idea, done in 82544d4c20c1f3bb9e3b9f521b75378d6f8144cd